File: Artio/Core/Entitites/Post.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Core.Entitites
{
    public class Post
    {
        public int PostId { get; set; }

        public Photo Image { get; set; }

        public string? Description { get; set; }

        public DateTimeOffset CreatedAt { get; set; }

        public string UserId { get; set; }

        public User User { get; set; }

        public List<PostTag> PostTags { get; set; } = new List<PostTag>();

        public List<Comment> Comments { get; set; } = new List<Comment>();

        public List<Like> Likes { get; set; } = new List<Like>();

    }
}

File: client-app/src/app/models/post.ts
import { Like } from "./like";
import { Photo } from "./photo";
import { Tag } from "./tag";
import { User } from "./user";

export interface Post {
    postId: number;
    image: Photo;
    description: string;
    createdAt: string;
    user: User;
    commentCount: number;
    tags: Tag[];
    likes: Like[];
}

export class PostFormValues {
    postId: number | undefined = undefined;
    image: Blob | null = null;
    description: string = "";
    tags: Tag[] = [];

    constructor(postId?: number,
        image?: Blob,
        description?: string,
        tags?: Tag[]) {
        if (postId && image && description && tags) {
            this.postId = postId;
            this.image = image;
            this.description = description;
            this.tags = tags;
        }
    }
}